planter/planter.sh

```diff
@@ -13,21 +13,44 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# NOTE: use like: ./planter.sh bazel build //cmd/...
-# NOTE: build kubernetes with: <path to test-infra>/planter.sh make bazel-build
+#
+# Use like: ./planter/planter.sh bazel build //cmd/...
+#
+# - $TAG can be overriden to choose a bazel version eg `TAG=0.8.0 planter.sh ...`
+# - $DOCKER_EXTRA can be set to supply extra args in the docker call
+# - $DRY_RUN will trigger echoing the docker call instead of running it
+# - planter will always `docker pull $IMAGE` before running it, see below
+#
+# To build kubernetes, first you checkout github.com/kubernetes/kubernetes
+# to $GOPATH/src/k8s.io/kubernetes and github.com/kubernetes/test-infra 
+# to $GOPATH/src/k8s.io/test-infra.
+#
+# Then you can build with:
+# $ cd $GOPATH/src/k8s.io/kubernetes`
+# $ $GOPATH/src/k8s.io/test-infra/planter/planter.sh make bazel-build`
+#
 
 set -o errexit
 set -o nounset
 IMAGE_NAME="gcr.io/k8s-testimages/planter"
 TAG="${TAG:-0.7.0}"
 IMAGE="${IMAGE_NAME}:${TAG}"
-# run our docker image as the host user with bazel cache and current repo dir
+# We want to mount our bazel workspace and the bazel cache
+# - WORKSPACE is assumed to be in your current git repo, or alternatively $PWD
 REPO=$(git rev-parse --show-toplevel 2>/dev/null || true)
 REPO=${REPO:-${PWD}}
-VOLUMES="-v ${REPO}:${REPO} -v ${HOME}:${HOME} --tmpfs /tmp:exec,mode=777"
+# - the bazel cache is in $HOME/.cache/bazel
+# - we could just mount $HOME/.cache/bazel, but unfortunately some rules
+# are still not terribly hermetic and actually need $HOME (rules_node)
+# - $HOME/.docker shouldn't be mounted so we clobber it with a tmpfs (issues/5736)
```

Comment:
what uses `$HOME/.docker`?

Reply:
you can have a `$HOME/.docker/config.json` for logging in to docker hub, which on mac will actually store credentials in your keychain which the container cannot access, this breaks rules_docker (see https://github.com/kubernetes/test-infra/issues/5736).
 
https://docs.docker.com/engine/reference/commandline/login/#extended-description

Reply:
can you update the comment to note this?

Reply:
done, also spaced out the bullets to hopefully make this more readable